mantidproject/mantid
# Refine goniometer angle and make consistent UBs

New function and unit test to make consistent UB required for common indexing of peaks across runs at different goniometer angles (i.e. no inversion or swapping of crystallographic axes). The goniometer axis and angle is found via the rotation matrix required to map saved UBs to the reference UB (first provided).

**To test:**
(1) Dowload the folder with the UBs [UBFiles.zip](https://github.com/mantidproject/mantid/files/4380599/UBFiles.zip)
(2) Add the folder with your UBs to the user directory list inmantid
(3) make a new folder inside the UB folder in which to save the consistent UBs and make this the default save directory
(4) Run the following script (if you are working from home make sure you're on the VPN as it searches for log files in the archive - it takes ~20s to read each log on my internet connection)
```
ubList = ['WISH00043350', 'WISH00043351', 'WISH00043352','WISH00043353','WISH00043354']
tab = FindGonioFromUB(UBfiles=",".join(ubList), chi=45, chiTol=5, phiTol=10, 
        phiHand=-1, dOmega=0, omegaHand=1)
```
Which should produce a table like this
![image](https://user-images.githubusercontent.com/55979119/77533496-e1260c80-6e8e-11ea-8bcd-0a8b849232a8.png)

The nominal value of chi =45 deg and the nominal log values for phi are
43350,     0
43351,   286
43352,   150
43353,    64
43354,    18

(5) Now that you know the phi/chi angles start playing around with the tolerances
(6) Play around with how the UBs are specified (should behave as Load does, e.g. if WISH is the default instrument the it is sufficient to just provide the mat files as e.g. '43350, 43351')
(7) make sure the documentation and pointers are clear

Fixes #28338 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
